test/distributed/fsdp/test_fsdp_optim_state.py

```diff
@@ -1882,6 +1883,32 @@ def step():
 
         self.run_subtests({"use_orig_params": [False, True]}, _run_test)
 
+    @skip_if_lt_x_gpu(2)
+    def test_use_orig_param_with_no_shard(self):
+        model = FSDP(
+            TestDummyModel().cuda(),
+            sharding_strategy=ShardingStrategy.NO_SHARD,
+            use_orig_params=True,
+        )
+        optim = torch.optim.Adam(model.parameters(), lr=1e-2)
+
+        def step():
```

Comment:
nit: Curious, why do we define `step()` here if we only call it once?

Reply:
copied from other unit tests:)

Reply:
I think it should be OK, it is neat to use a function